### PR TITLE
feat: Added PR template.

### PR DESCRIPTION
**Pull request by andre-at-situ**
How tested?

This is the test lolz :)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## Type of change
+- [ ] Bug fix (non-breaking change which fixes a bug)
+- [ ] Chore (non-breaking change that refactors, tidies up or formats code)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
```
Comment (willsawyerrrr) [inline .github/pull_request_template.md:5]: This information is conveyed through a label on the pull request and its associated issue (if any).

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+If the change is breaking, describe the actions taken to update those packages/repos that depend on this repo:
```
Comment (willsawyerrrr) [inline .github/pull_request_template.md:7]: This is only relevant to libraries. These actions should be described in their release notes, not in a pull request.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+## How Has This Been Tested?
+Please describe the tests that you ran to verify your changes. Provide instructions so test can be reproduced.
+
+## Checklist:
+- [ ] All build checks are passing
```
Comment (willsawyerrrr) [inline .github/pull_request_template.md:13]: This requirement is enforced by status checks.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
```
Comment (willsawyerrrr) [inline .github/pull_request_template.md:15]: Locally doesn't matter, all unit tests need to pass within the status check on this PR.

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+If the change is breaking, describe the actions taken to update those packages/repos that depend on this repo:
+
+## How Has This Been Tested?
+Please describe the tests that you ran to verify your changes. Provide instructions so test can be reproduced.
+
+## Checklist:
+- [ ] All build checks are passing
+- [ ] I have added tests that prove my fix is effective or that my feature works
```
Comment (willsawyerrrr) [inline .github/pull_request_template.md:14]: This will be conveyed through changes in the test directory as part of this PR.

**Comment by willsawyerrrr** [comment]
I'm not entirely opposed to a pull request template, though I think it needs to convey better information than this suggestion.

**Comment by willsawyerrrr** [comment]
For consistency with issue templates, any pull request template(s) we create should be under the `.github/PULL_REQUEST_TEMPLATE` directory with meaningful names, not just within `.github`.

File: .github/pull_request_template.md (hunk 1 above)
Comment (andre-at-situ) [reply .github/pull_request_template.md:5]: How does the label tell me then change is breaking and/or if the author has thought about it?

File: .github/pull_request_template.md (hunk 2 above)
Comment (andre-at-situ) [reply .github/pull_request_template.md:7]: For a formal library that we don't publish I agree.. but if one of us updates an internal library to effective use that change in another repo straight-away, then this information is useful.

File: .github/pull_request_template.md (hunk 3 above)
Comment (andre-at-situ) [reply .github/pull_request_template.md:13]: This is a checklist. It means the author knows they should check all of the status checks.

**Comment by andre-at-situ** [comment]
> For consistency with issue templates, any pull request template(s) we create should be under the `.github/PULL_REQUEST_TEMPLATE` directory with meaningful names, not just within `.github`.

I don't even know if this going to work - file locations don't need to be precise for an experiment.

File: .github/pull_request_template.md (hunk 5 above)
Comment (andre-at-situ) [reply .github/pull_request_template.md:14]: This is a checklist. It's to remind the author that they need to add tests and test their feature.

**Comment by andre-at-situ** [comment]
> I'm not entirely opposed to a pull request template, though I think it needs to convey better information than this suggestion.

What do you think should be included then? What information do you want to see?

File: .github/pull_request_template.md (hunk 4 above)
Comment (andre-at-situ) [reply .github/pull_request_template.md:15]: This is a checklist and informs the reviewer that the author has thought about and has conducted testing.

File: .github/pull_request_template.md (hunk 2 above)
Comment (willsawyerrrr) [reply .github/pull_request_template.md]: Why should our internal libraries be any less formally managed and released?

Comment (andre-at-situ) [reply .github/pull_request_template.md]: That wasn't what I was suggesting. The publication of release notes is a separate issue to what I've brought up here.

If someone makes a change to an internal library then I see it as their responsibility to fix and/or create issues for all the things they may break in dependent packages.

We could just address this by limiting version bumps to point releases but this isn't being very friendly to the person updating the code that depends on the library.

If someone is making a company wide library change then I'd like to see that they're thinking about the implications of the change. This is especially important with minimal test coverage.

File: .github/pull_request_template.md (hunk 3 above)
Comment (andre-at-situ) [reply .github/pull_request_template.md]: I updated this btw. The author only knows code checks are passing once the PR is created so its unfair to ask if they pass here.

File: .github/pull_request_template.md (hunk 1 above)
Comment (willsawyerrrr) [reply .github/pull_request_template.md]: Can you please explain what you mean by "breaking change" in this context?

My mind goes straight to a major version release of a product or library, which isn't relevant to most of our repositories: I think it'd only be relevant on the Automation suite, the dev container configurations, the Entegy SDK and the FastAPI starter.

If you simply mean anything which changes how the application works, that would include all fixes and features.

**Review by willsawyerrrr — DISMISSED** [review]
Please move to the appropriate directory for extensibility as per https://github.com/SituDevelopment/.github/pull/58#issuecomment-1916286601. Otherwise, good to go.

**Review by willsawyerrrr — CHANGES_REQUESTED** [review]
https://github.com/SituDevelopment/.github/pull/58#pullrequestreview-1869567083

**Pull request merged**